### PR TITLE
Added FillingContainer class.

### PR DESCRIPTION
**Pull request by raonygamer**
Added FillingContainer class to be able to work with filling containers (chests/shulker boxes/ender chests) etc

### CODE REVIEW

File: AmethystAPI/src/minecraft/src/common/world/inventory/FillingContainer.cpp [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+#include "minecraft/src/common/world/item/ItemStack.hpp"
+
+extern "C" void* FillingContainer_vtable;
+
+FillingContainer::FillingContainer(Player* player, int size, ContainerType type) : Container(type),
+    mPlayer(player)
+{
+    //workaround to place the original game vtable on the class instance even in a implemented constructor
+    *reinterpret_cast<void**>(this) = FillingContainer_vtable;
```
Comment (FrederoxDev) [inline AmethystAPI/src/minecraft/src/common/world/inventory/FillingContainer.cpp:11]: Why is this needed? If you're planning to construct from this, doing asm definitions for this should be good?

Comment (raonygamer) [reply AmethystAPI/src/minecraft/src/common/world/inventory/FillingContainer.cpp:11]: forgot to remove xD
